### PR TITLE
change pytorch package to cpu only

### LINKED ISSUE #3043: Why nvidia is packed into the frontend container?
image

### PR DESCRIPTION
**Pull request by charliez0**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**

fixes #3043 
change pytorch package to cpu only as gpu function not needed in this package
PS: This change can reduce 50% of the image size!!!
---

**Give a summary of what the PR does, explaining any non-trivial design decisions**

---
**Other references**

### CODE REVIEW

**Review by neubig — COMMENTED** [review]
This seems like a really good idea to me! I'd like to ask @xingyaoww for a double-check to make sure I'm not missing anything though.

**Comment by neubig** [comment]
Ah, and it seems like tests are also not passing... @charliez0 , if you could take a look at the test errors and see if there's a way to resolve them that'd be great!

**Comment by charliez0** [comment]
> Ah, and it seems like tests are also not passing... @charliez0 , if you could take a look at the test errors and see if there's a way to resolve them that'd be great!

@neubig updated lock file, it should be working now

**Comment by charliez0** [comment]
oh, pytorch does not provide official binarys, I need to use other source, wait a moment

**Comment by xingyaoww** [comment]
agree! I remember the pytorch dependency comes from llama-index.. and I'm not quite sure where we are using it for - maybe we could simplify the dependency (install a version of llama-index without pytorch - we only use API embedding models) there?

**Comment by charliez0** [comment]
> agree! I remember the pytorch dependency comes from llama-index.. and I'm not quite sure where we are using it for - maybe we could simplify the dependency (install a version of llama-index without pytorch - we only use API embedding models) there?

indeed by llama-index-embeddings-huggingface, can we skip this package?

**Comment by xingyaoww** [comment]
@charliez0 Let's try and see if it breaks anything 🤣 if not - it is probably fine. I don't think using embedding is an essential feature at the current stage, so make the codebase/dependency simpler is probably a good choice for now.

**Comment by charliez0** [comment]
> @charliez0 Let's try and see if it breaks anything 🤣 if not - it is probably fine. I don't think using embedding is an essential feature at the current stage, so make the codebase/dependency simpler is probably a good choice for now.
> 
> EDIT: we could also make `llama-index-embeddings-huggingface` an optional dependency, e.g., you can install it by running smth like `make build-optional`

we can include only core and some basic, but idk which to choose

**Comment by charliez0** [comment]
```plain
Quickstart Installation from Pip[#](https://docs.llamaindex.ai/en/stable/getting_started/installation/#quickstart-installation-from-pip)
To get started quickly, you can install with:


pip install llama-index
This is a starter bundle of packages, containing

llama-index-core
llama-index-legacy # temporarily included
llama-index-llms-openai
llama-index-embeddings-openai
llama-index-program-openai
llama-index-question-gen-openai
llama-index-agent-openai
llama-index-readers-file
llama-index-multi-modal-llms-openai
NOTE: LlamaIndex may download and store local files for various packages (NLTK, HuggingFace, ...). Use the environment variable "LLAMA_INDEX_CACHE_DIR" to control where these files are saved.
```

**Comment by xingyaoww** [comment]
> idk which to choose

Free free to find relevant usage of `llama-index` in the codebase and see what is actually used now..

For things that are not used yet (or less frequently used), i think you can just remove that from the dependency, and add a check around the import code (e.g., if import fails, raise an error message to tell the user you should "poetry install XXX" or run `make build-optional`)

File: poetry.lock
Comment (yufansong) [inline poetry.lock:1]: You update too much poetry lock. Please revert it and only update the neccessary one.

**Comment by tobitege** [comment]
@charliez0 I'm sorry, but I have to close this PR. The problem is, that with the "+cpu" naming of the `torch` package our MacOSX tests (which run poetry then) will fail and I tried different conditions in the `pyproject.toml` file to differentiate the os for this.

**Pull request closed without merging**